database.py

```diff
@@ -1,38 +1,38 @@
 #!/usr/bin/env python3
 """Initialize the mailgun-relayery channel database."""
-import argparse
 import pathlib
-import os
 
+import icontract
 import lmdb
 
 # Descriptor -> Channel database
-DB_CHANNEl_KEY = 'channelDb'.encode()  # database name
+DB_CHANNEL_KEY = 'channelDb'.encode()  # database name
```

Comment:
channel and timestamp db must reside in the same environment -- you also need to delete the timestamp if you delete the channel (in the same transaction).

Reply:
make NewTransaction accept ``access``, where ``access`` is an enum (`Control`, `Relay` or whatever is more suitable); add preconditions so that some functions can be called by both, but some functions on transaction can be only called by one or the other.